components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoPermissionMigrationFacts.kt

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.permission
+
+import mozilla.components.support.base.Component
+import mozilla.components.support.base.facts.Action
+import mozilla.components.support.base.facts.Fact
+import mozilla.components.support.base.facts.collect
+
+/**
+ * Facts emitted for telemetry related to site permissions migration.
+ */
+class GeckoPermissionMigrationFacts {
+
+    /**
+     * Items that specify a permission was migrated from
+     * the local storage to the gecko view storage.
+     */
+    object Items {
+        const val PERMISSION_MIGRATED = "permission_migrated"
+    }
+}
+
+private fun emitMigrationFact(
+    action: Action,
+    item: String,
+    value: String
+) {
+    Fact(
+        Component.BROWSER_ENGINE_GECKO,
+        action,
+        item,
+        value
+    ).collect()
+}
+
+internal fun emitPermissionMigratedFact(value: String) = emitMigrationFact(
+    Action.INTERACTION,
+    GeckoPermissionMigrationFacts.Items.PERMISSION_MIGRATED,
+    value
+)
```

Comment:
There may be a lot of individual permission migrations happening. I wonder if it needs to be this fine-grained? Maybe just having a "not migrated permissions" count per user is good enough and shows us how far along we are in the migration?

Reply:
For doing that we will need to do a spesific query to extract that information, that is not part of the normal user flow,
 maybe we could take the telemetry part as follow up? An created a spesific that provide us that status. We could query that status and report it on startup.

Reply:
Yeah, +1: Let's do telemetry as a follow-up. We can add this "count" at any time and it wouldn't need to ship in the first version.

Reply:
I filed https://github.com/mozilla-mobile/android-components/issues/10296 for it